src/Tasks/Microsoft.NET.Build.Tasks/ResolvePackageDependencies.cs

```diff
@@ -399,6 +403,56 @@ private void GetFileDependencies(LockFileTargetLibrary package, string targetNam
             }
         }
 
+        private void GetDependencyDiagnostics()
+        {
+            Dictionary<string, string> projectDeps = LockFile.GetProjectFileDependencies();
+
+            // if a project dependency is not in the list of libs, then it is an unresolved reference
+            var unresolvedDeps = projectDeps.Where(dep =>
+                null == LockFile.Libraries.FirstOrDefault(lib =>
+                    string.Equals(lib.Name, dep.Key, StringComparison.OrdinalIgnoreCase)));
+
+            foreach (var target in LockFile.Targets)
+            {
+                foreach (var dep in unresolvedDeps)
+                {
+                    string packageId = $"{dep.Key}/{dep.Value}";
+
+                    Diagnostics.Add(nameof(Strings.NU1001),
+                        string.Format(CultureInfo.CurrentCulture, Strings.NU1001, dep),
+                        ProjectPath,
+                        DiagnosticMessageSeverity.Warning,
+                        1, 0,
+                        target.Name,
+                        packageId,
+                        logToMSBuild: false // Temporarily suppress MSBuild logging because this diagnostic is also reported by NuGet
```

Comment:
Is there a bug tracking this on nuget side + one on our side to follow up when that's done?